sniffles/petabi_rule_writer/petabi_rule_writer.py

```diff
@@ -0,0 +1,336 @@
+        except Exception as err:
+            print("Could not read regex file")
+            print("regexParser: " + str(err))
+            raise Exception("The program will stop.")
+        line = fd.readline()
+        while line:
+            myregex = line.strip()
+            # Check if content is in right format
+            if not check_pcre_compile(myregex):
+                print("Error: Unknown regex format")
+                print("Regex: \"" + myregex + "\" will be ignored")
+                line = fd.readline()
+                continue
+            regex.append(myregex)
+            line = fd.readline()
+        fd.close()
+    return regex
+
+
+def check_pcre_compile(re):
+    optiosn = []
+    if len(re) and re[0] == '/':
+        optp = re.rfind('/')
+        if optp > 0:
+            options = list(re[optp + 1:])
+            re = re[1:optp]
+    opts = 0
+    for opt in optiosn:
+        if opt in PCRE_OPT:
+            opts |= PCRE_OPT[opt]
+    try:
+        sniffles.pcrecomp.compile(re, opts)
+    except:
+        return False
+    return True
+
+
+def formatTrafficStreamRule(proto='tcp', src='any', dst='any', dport='any',
+                            sport='any', trafficAck=False,
+                            out_of_order=False, out_of_order_prob=False,
+                            packet_loss=False, tcpOverlap=False):
+    trafficStreamFormat = []
+    header = "    <traffic_stream"
+    trafficStreamFormat.append(header)
+    tail = ">"
+    protoFormat = "proto=\"" + proto + "\""
+    trafficStreamFormat.append(protoFormat)
+    srcFormat = "src=\"" + src + "\""
+    trafficStreamFormat.append(srcFormat)
+    dstFormat = "dst=\"" + dst + "\""
+    trafficStreamFormat.append(dstFormat)
+    dportFormat = "dport=\"" + dport + "\""
+    trafficStreamFormat.append(dportFormat)
+    sportFormat = "sport=\"" + sport + "\""
+    trafficStreamFormat.append(sportFormat)
+
+    if trafficAck:
+        myAck = "ack=\"true\""
+        trafficStreamFormat.append(myAck)
+    if out_of_order:
+        out_of_order_format = "out_of_order=\"true\""
+        trafficStreamFormat.append(out_of_order_format)
+    if out_of_order_prob:
+        oopFormat = "out_of_order_prob=\"" + out_of_order_prob + "\""
+        trafficStreamFormat.append(oopFormat)
+    if packet_loss:
+        packet_loss_format = "packet_loss=\"" + packet_loss + "\""
+        trafficStreamFormat.append(packet_loss_format)
+    if tcpOverlap:
+        tcpOverlapFormat = "tcp_overlap=\"true\""
+        trafficStreamFormat.append(tcpOverlapFormat)
+    trafficStream = ' '.join(trafficStreamFormat)
+    trafficStream = trafficStream + tail + "\n"
+    return trafficStream
+
+
+def formatPktRule(regexList=None, count='1', fragment=False,
+                  flow='to server', split=False, ttl=None, ttlExpiry=False,
+                  pktAck=False):
+    pktRule = []
+    count = "times=\"" + count + "\""
+    flowOpt = flow
+    if regexList:
+        head = "<pkt"
```

Comment:
I'm not sure much benefit is gained from creating a variable for head.  Just add head first to the pkt rule, and finish with tail.  I don't feel very strong about it though.